lib/swiftype-app-search/client/documents.rb

```diff
@@ -19,6 +19,22 @@ def get_documents(engine_name, ids)
         get("engines/#{engine_name}/documents", ids)
       end
 
+      # Index a document using the {App Search API}[https://swiftype.com/documentation/app-search/].
+      #
+      # @param [String] engine_name the unique Engine name
+      # @param [Array] document a Document Hash
+      #
+      # @return [Bool] true when the request succeeds
+      #
+      # @raise [SwiftypeAppSearch::InvalidDocument] when the document is missing required fields, contains unsupported fields, or has processing errors returned from the api
+      # @raise [Timeout::Error] when timeout expires waiting for statuses
+      def index_document(engine_name, document)
+        response = index_documents(engine_name, [document])
+        errors = response.first['errors']
+        raise InvalidDocument.new(errors.join('; ')) if errors.any?
+        true
```

Comment:
This should still return the body of the response (possibly without `'errors'` since we know it's always empty).  Statically returning `true` is not very useful, might as well just return `nil`. And by returning a `Hash` we can later include other details the API returns in the future without breaking client api compatbility.